sdk/client/src/broadcast_client.rs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    error::WaitForTransactionError,
+    views::{AccountView, MetadataView, TransactionView},
+    Client, Error, MethodRequest, MethodResponse, Response, Result,
+};
+use diem_crypto::hash::CryptoHash;
+use diem_types::{
+    account_address::AccountAddress,
+    transaction::{SignedTransaction, Transaction},
+};
+
+use crate::views::{AccountStateWithProofView, CurrencyInfoView, EventView, StateProofView};
+use futures::future::join_all;
+use rand::seq::{SliceChooseIter, SliceRandom};
+use std::time::Duration;
+
+#[derive(Clone, Debug)]
+struct BroadcastingClient {
+    pub clients: Vec<Client>,
+    pub num_parallel_requests: usize,
+}
+
+impl BroadcastingClient {
+    pub fn new(clients: Vec<Client>, num_parallel_requests: usize) -> Result<Self, String> {
+        if num_parallel_requests < 1 {
+            return Err("num_parallel_requests should be >= 1".into());
+        }
+        if num_parallel_requests > clients.len() {
+            return Err(format!(
+                "num_parallel_requests({}) should be <= length of clients({})",
+                num_parallel_requests,
+                clients.len()
+            ));
+        }
+        Ok(Self {
+            clients,
+            num_parallel_requests,
+        })
+    }
+
+    pub async fn wait_for_signed_transaction(
+        &self,
+        txn: &SignedTransaction,
+        timeout: Option<Duration>,
+        delay: Option<Duration>,
+    ) -> Result<Response<TransactionView>, WaitForTransactionError> {
+        self.wait_for_transaction(
+            txn.sender(),
+            txn.sequence_number(),
+            txn.expiration_timestamp_secs(),
+            &Transaction::UserTransaction(txn.clone()).hash().to_hex(),
+            timeout,
+            delay,
+        )
+        .await
+    }
+
+    pub async fn wait_for_transaction(
+        &self,
+        address: AccountAddress,
+        seq: u64,
+        expiration_time_secs: u64,
+        txn_hash: &str,
+        timeout: Option<Duration>,
+        delay: Option<Duration>,
+    ) -> Result<Response<TransactionView>, WaitForTransactionError> {
```

Comment:
So this is sort of messy...I'm going to make an executive decision and say that you should just drop the wait_for functions for now. If AOS needs them we can add them later.